### ISSUE TITLE
no-unused-variable in callback function gets reported but should not

### ISSUE THREAD

**Issue by maciossek** [palantir/tslint#578]
this reports an error:

```typescript
private toggleSwitch(isPrevAnimation: boolean): void {
  if (isPrevAnimation) {
    this.modalInnerElement.addClass('transform-left');
    this.$animate.addClass(this.modalInnerElement, 'transform-right').then(function() {
      this.removeOldMenu();
    }.bind(this));
  } else {
    this.$animate.addClass(this.modalInnerElement, 'transform-left').then(function() {
      this.modalInnerElement.removeClass('transform-left');
      this.removeOldMenu();
    }.bind(this));
  }
}

private removeOldMenu(): void {
  var oldMenuElement: ng.IAugmentedJQuery;
  if (this.openedMenus.length > 1) {
    oldMenuElement = this.openedMenus.shift();
    oldMenuElement.scope().$destroy();
    oldMenuElement.remove();
    this.modalInnerElement.removeClass('transform-left');
    this.modalInnerElement.removeClass('transform-right');
  }
}
```

`unused variable: 'removeOldMenu'`

**Comment by adidahiya** [comment]
this sounds like a limitation of the compiler. typescript doesn't do any special handling of `Function.prototype.bind`, so it doesn't know the `this` in `this.removeOldMenu()` references the class.

I'd suggest a fat arrow function instead of `Function.prototype.bind`:

```ts
    this.$animate.addClass(this.modalInnerElement, 'transform-right').then(() => {
      this.removeOldMenu();
    });
```

**Comment by maciossek** [comment]
Yup, that did the trick. Thanks!
